appstudio-controller/controllers/appstudio.redhat.com/deploymenttargetreclaimer_controller.go

```diff
@@ -89,62 +104,84 @@ func (r *DeploymentTargetReconciler) Reconcile(ctx context.Context, req ctrl.Req
 		log.Info("Added finalizer to DeploymentTarget", "finalizer", FinalizerDT)
 	}
 
+	if dt.DeletionTimestamp == nil {
+		// The DeploymentTarget is not currently being deleted, so no more work to do.
+		return ctrl.Result{}, nil
+	}
+
 	// Handle deletion if the DT has a deletion timestamp set.
-	if dt.GetDeletionTimestamp() != nil {
-		var sr *codereadytoolchainv1alpha1.SpaceRequest
-		if sr, err = findMatchingSpaceRequestForDT(ctx, r.Client, &dt); err != nil {
-			if !apierr.IsNotFound(err) {
-				return ctrl.Result{}, err
-			}
-		}
-		if sr == nil {
-			if removeFinalizer(&dt, FinalizerDT) {
-				if err := r.Client.Update(ctx, &dt); err != nil {
-					return ctrl.Result{}, fmt.Errorf("failed to remove finalizer %s from DeploymentTarget %s in namespace %s: %v", FinalizerDT, dt.Name, dt.Namespace, err)
-				}
-				log.Info("Removed finalizer from DeploymentTarget", "finalizer", FinalizerDT)
-				return ctrl.Result{}, nil
-			}
+	var sr *codereadytoolchainv1alpha1.SpaceRequest
```

Comment:
The rest of this function is just refactoring: the logic itself should not have changed.